S3/Config.py

```diff
@@ -133,6 +133,8 @@ class Config(object):
     force = False
     server_side_encryption = False
     enable = None
+    # Used to allow colons in bucket names for Ceph compatibility
+    bucket_name_quirks = False
```

Comment:
I don't like so much "quirks" and to restrict that to colons and Ceph.

Maybe you can call that something like: "bucket_name_extended",  "extended_bucket_name", "bucket_name_relaxed", "bucket_name_lenient".

or invert the condition like:
"bucket_name_strict = True", 

And in the comment say that it relax rules for bucket name for S3 compatible services that supports path style requests.